### PR TITLE
V4

### PR DESCRIPTION
**Pull request by iefserge**
- enforce ES6 import / exports
- enforce display names on react components
- enforce presence of keys on react iterator children
- enforce presence of dangling commas

T = major
R = @micnews/engineers

### CODE REVIEW

File: eslintrc.json [hunk 1]
```diff
@@ -5,19 +5,23 @@
...
+    "react/display-name": [2, { "ignoreTranspilerName": true }],
     "react/jsx-boolean-value": [2, "always"],
+    "react/jsx-filename-extension": [0],
+    "react/jsx-key": 2,
     "react/jsx-no-duplicate-props": 2,
-
-    "generator-star-spacing": 0,
-    "babel/generator-star-spacing": [2, { "before": false, "after": true }]
+    "react/no-direct-mutation-state": 2,
+    "react/require-extension": [0]
```
Comment (iefserge) [inline eslintrc.json:23]: why disable this rule?

File: eslintrc.json [hunk 2]
```diff
@@ -5,19 +5,23 @@
...
+    "import/no-commonjs": 2,
 
     "jsx-a11y/label-has-for": [0],
 
-    "react/require-extension": [0],
-    "react/jsx-filename-extension": [0],
-    "react/no-direct-mutation-state": 2,
+    "react/display-name": [2, { "ignoreTranspilerName": true }],
     "react/jsx-boolean-value": [2, "always"],
+    "react/jsx-filename-extension": [0],
```
Comment (iefserge) [inline eslintrc.json:19]: disabled?

File: eslintrc.json (hunk 1 above)
Comment (danmakenoise) [reply eslintrc.json:23]: I didn't change it just sorted it.

File: eslintrc.json (hunk 2 above)
Comment (danmakenoise) [reply eslintrc.json:19]: Just reordered them.

**Pull request merged**